nservicebus/upgrades/5to6/distributor.md

```diff
@@ -0,0 +1,94 @@
+---
+title: Migrating the Distributor to use Sender Side Distribution
+tags:
+ - upgrade
+ - migration
+related:
+ - samples/scaleout/distributor-upgrade
+---
+
+The [distributor](/nservicebus/scalability-and-ha/distributor) has been deprecated and replaced by [sender-side distribution](/nservicebus/msmq/scalability-and-ha/sender-side-distribution.md) with NServiceBus Version 6.
+
+## Client-side distribution
+
+With sender-side distribution, senders are aware of scaled out receivers, allowing senders to distribute messages directly to the receivers instead of using an intermediary distributor.
+
+
+## Upgrade scenario
+
+This process aims to allow upgrade in production environments without missing events and minimal downtime.
+
+* Upgrade all endpoints that interact with the distributor to Version 6 first. For example, upgrade all endpoints that send command messages to the Distributor endpoint for the load to be distributed, or endpoints that send subscription messages to the distributor, subscribing to events when a worker node publishes an event. At this stage, do not upgrade the workers to Version 6.
+* Apply the following steps for each scaled out instance enlisted to the Distributor, one instance after another:
+  * Shut down the instance.
+  * [Upgrade the instance to NServiceBus Version 6](#upgrade-endpoint-to-version-6).
+  * [Enlist it with the distributor](#enlist-version-6-endpoints-with-a-distributor).
+  * Start the instance again.
+* Configure [sender-side distribution](/nservicebus/msmq/scalability-and-ha/sender-side-distribution.md) for all endpoints sending commands or receiving events from the distributor.
+* Detach the instances from the Distributor by  applying the following steps to the instances enlisted to the Distributor. But *skip this step for at least one instance* to ensure some instances are still attached to the distributor.
+  * Shut down the instance.
+  * Detach the instance from the Distributor by removing the `EnlistWithLegacyMSMQDistributor` configuration.
+  * Start the instance again.
+
+Most instances should now be detached from the Distributor. The Distributor can now either be upgraded to a regular worker node or be decommissioned:
+
+### Upgrade the Distributor to a worker node
+
+This approach allows you to continue using the resources used for the Distributor as a regular worker node.
+
+* Endpoint need to have the same endpoint name as other workers. Change queue somehow?
```

Comment:
I added that as a reminder to think the scenario through where the distributor has a different name than the workers, which means that the input queue is different too. But since we now support a "queue" attribute in the instance mapping file, I think this shouldn't be an issue and I can remove that point. Maybe @SzymonPobiega can confirm?